File: api/Ganymede.Api.Models/App/FormData/SpellFormData.cs
using Ganymede.Api.Data.Spells;
using Ganymede.Api.Models.Spells;
using System.Collections.Generic;
using static Ganymede.Api.Models.Spells.Enums;

namespace Ganymede.Api.Models.App.FormData
{
    public class SpellFormData
    {
        public List<SpellSchoolModel> Schools { get; set; }
        public List<CastingTimeType> CastingTimeTypes { get; set; }
        public List<string> CastingTimeUnits { get; set; }
        public List<RangeType> RangeTypes { get; set; }
        public List<string> RangeUnits { get; set; }
        public List<string> RangeShapes { get; set; }
        public List<string> DurationUnits { get; set; }
        public List<DurationType> DurationTypes { get; set; }
    }
}

File: src/app/models/core/app/forms/formData/spellFormData.ts
import { SpellSchool } from "../../../../../campaign/modules/spell/models/spellSchool";

export class SpellFormData {
	public schools: SpellSchool[];
	public castingTimeUnits: string[];
	public rangeTypes: string[];
	public rangeUnits: string[];
	public rangeShapes: string[];
	public durations: string[];
	public durationTypes: string[];

	static getDefault(): SpellFormData {
		return {
			schools: [],
			castingTimeUnits: [],
			rangeTypes: [],
			rangeUnits: [],
			rangeShapes: [],
			durations: [],
			durationTypes: []
		};
	}
}
